src/lib/support/Pool.h

```diff
@@ -660,4 +660,39 @@ class ObjectPool<T, N, ObjectPoolMem::kHeap> : public HeapObjectPool<T>
 };
 #endif // CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
 
+/// RAII class for pool allocation that guarantees that ReleaseObject() will be called.
+/// This is effectively a simple unique_ptr, except calling pool.ReleaseObject instead of delete
+template <typename T, typename PoolT>
+class PoolAutoRelease
```

Comment:
Unit tests and separate PR?

Reply:
Addressed @ https://github.com/project-chip/connectedhomeip/pull/40310